crates/yttrium/src/uniffi_compat/stacks.rs

```diff
@@ -281,6 +250,97 @@ impl StacksClient {
+        &self,
+        wallet: &str,
+        network: &str,
+        request: TransferStxRequest,
+    ) -> Result<TransferStxResponse, StacksSignTransactionError> {
+        let mut stacks_wallet = StacksWallet::from_secret_key(wallet)
+            .map_err(StacksSignTransactionError::InvalidSecretKey)?;
+        let sender_key = stacks_wallet
+            .get_account(0)
+            .map_err(StacksSignTransactionError::UnwrapPrivateKey)?
+            .private_key()
+            .map_err(StacksSignTransactionError::UnwrapPrivateKey)?;
+
+        let account = match self.get_account(network, &request.sender).await {
+            Ok(result) => result,
+            Err(e) => {
+                let error_string =
+                    format!("Fetch account failed. Error: {}", e);
+                return Err(StacksSignTransactionError::FetchAccount(
+                    error_string,
+                ));
+            }
+        };
+
+        // Use fallback fee rate of 180 if transfer_fees fails
+        let fee_rate = match self.transfer_fees(network).await {
+            Ok(result) => result,
+            Err(e) => {
+                println!("Failed to fetch transfer fee rate: {}. Using fallback value of 1 fee rate / byte.", e);
+                1
+            }
+        };
+
+        // https://github.com/52/stacks.rs/blob/c6455fe5eeae04b2f0e3a88fe6e6c803949a8417/README.md?plain=1#L24
+        let mut transfer = match network {
+            network::stacks::MAINNET => STXTokenTransfer::builder()
+                .recipient(clarity!(PrincipalStandard, request.recipient))
+                .amount(request.amount)
+                .memo(request.memo)
+                .sender(sender_key)
+                .network(StacksMainnet::new())
+                .build()
+                .transaction(),
+            network::stacks::TESTNET => STXTokenTransfer::builder()
+                .recipient(clarity!(PrincipalStandard, request.recipient))
+                .amount(request.amount)
+                .memo(request.memo)
+                .sender(sender_key)
+                .network(StacksTestnet::new())
+                .build()
+                .transaction(),
+            _ => {
+                return Err(StacksSignTransactionError::InvalidNetwork(
+                    network.to_string(),
+                ))
+            }
+        };
+
+        let nonce = account.nonce;
+        transfer.set_nonce(nonce);
+
+        let bytes_length = clarity::Codec::encode(&transfer)
+            .map_err(StacksSignTransactionError::Encode)?
+            .len();
+        let mut fees = bytes_length as u64 * fee_rate;
+        // fees can't be lower than 180 microSTX
+        // A typical single-signature STX transfer is ~180–200 bytes and lower fee_rate is 1
+        // Fee is calculated as fee_rate * transaction_bytes
+        // Anything below this will often fail with "FeeTooLow"
+        fees = std::cmp::max(fees, 180); 
+        transfer.set_fee(fees);
```

Comment:
Avoid mutable variables unless you actually need them. Also consider using a scope to avoid accidentally using the wrong value later

```suggestion
        {
            let fees = bytes_length as u64 * fee_rate;
            // fees can't be lower than 180 microSTX
            // A typical single-signature STX transfer is ~180–200 bytes and lower fee_rate is 1
            // Fee is calculated as fee_rate * transaction_bytes
            // Anything below this will often fail with "FeeTooLow"
            let capped_fees = std::cmp::max(fees, 180); 
            transfer.set_fee(capped_fees);
        }
```